homeassistant/components/sensor/yweather.py

```diff
@@ -130,12 +139,22 @@ def entity_picture(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
+        if self._day and self._date:
+            return {
```

Comment:
Instead of copying the dict between the two different returns, only extend the object when `day` and `date` are set..